src/server/shared/Packets/ByteBuffer.h

```diff
@@ -203,20 +203,22 @@ class TC_SHARED_API ByteBuffer
             return *this;
         }
 
-        ByteBuffer &operator<<(const std::string &value)
+        ByteBuffer &operator<<(std::string_view value)
         {
             if (size_t len = value.length())
-                append((uint8 const*)value.c_str(), len);
-            append((uint8)0);
+                append(reinterpret_cast<uint8 const*>(value.data()), len);
+            append(static_cast<uint8>(0));
             return *this;
         }
 
-        ByteBuffer &operator<<(const char *str)
+        ByteBuffer& operator<<(std::string const& str)
         {
-            if (size_t len = (str ? strlen(str) : 0))
-                append((uint8 const*)str, len);
-            append((uint8)0);
-            return *this;
+            return operator<<(std::string_view(str));
+        }
+
+        ByteBuffer &operator<<(char const* str)
+        {
+            return operator<<(std::string_view(str));
```

Comment:
old code handled nullptr safely, this does not

Reply:
it did not - old code called `append(nullptr, 0)` which would assert out